### PR TITLE
Optimize sync script

### PR DESCRIPTION
**Pull request by LongfeiH**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] [local build the book](https://github.com/ocademy-ai/machine-learning/blob/main/CONTRIBUTING.md#build-the-book)
- [ ] review the artifacts from local build
- [ ] [local build the slides](https://github.com/ocademy-ai/machine-learning/blob/main/CONTRIBUTING.md#build-the-slides-optional) if appliable
- [x] Unit test if appliable

# Checklist:

- [x] My code follows the [style guidelines](https://github.com/ocademy-ai/machine-learning/blob/main/STYLE_GUIDE.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have applied tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/pr-awesome.yml [hunk 1]
```diff
@@ -28,5 +28,10 @@ jobs:
         sqlite3 target_data.db .dump > target_data.dump
       continue-on-error: true
       
-    - name: Show differences
+    - name: Show database differences
       run: diff -a target_data.dump current_data.dump || true
+
+    - name: Synchronize local and cloud databases
```
Comment (zhangqi444) [inline .github/workflows/pr-awesome.yml:34]: Please don't add such steps to the action. You should create master-awesome and main-awesome and add the backend sync step. Also, the main sync should target to test env, while release should target to prod env.

Comment (LongfeiH) [reply .github/workflows/pr-awesome.yml]: I have made corresponding changes in the new commit, please check it.

File: .github/workflows/pr-main-awesome.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+name: pr-main-awesome
+
+on:
+  pull_request:
```
Comment (zhangqi444) [inline .github/workflows/pr-main-awesome.yml:4]: You should not monitor pr for publishing data. Instead, you should listen to code push. Please refer to main.yml.

Comment (LongfeiH) [reply .github/workflows/pr-main-awesome.yml]: I have changed the pull_request to push in two yml file, please check it.

File: .github/workflows/pr-main-awesome.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
+name: pr-main-awesome
```
Comment (zhangqi444) [inline .github/workflows/pr-main-awesome.yml:1]: Please fix the name and the file name as well.

Comment (LongfeiH) [reply .github/workflows/pr-main-awesome.yml]: done

**Review by zhangqi444 — APPROVED** [review]

**Pull request merged**